src/world/FieldComputations.cpp

```diff
@@ -185,125 +94,113 @@ std::shared_ptr<Vector2> FieldComputations::lineIntersectionWithDefenceArea(cons
+        double distanceSecondIntersection = lineStart.dist(intersections.at(1));
+        return std::make_shared<Vector2>(distanceFirstIntersection < distanceSecondIntersection ? intersections.at(0) : intersections.at(1));
+    } else {
+        return nullptr;
     }
-    return nullptr;
 }
 
 Polygon FieldComputations::getDefenseArea(const Field &field, bool ourDefenseArea, double margin, bool includeOutSideField) {
-    double backLineUsXCoordinate = includeOutSideField ? field.getLeftmostX() - field.getBoundaryWidth() : field.getLeftmostX() - margin;
-    double backLineThemXCoordinate = includeOutSideField ? field.getRightmostX() + field.getBoundaryWidth() : field.getRightmostX() + margin;
-
-    std::vector<Vector2> defenceAreaUsPoints = {{field.getLeftPenaltyLine().begin.x + margin, field.getLeftPenaltyLine().begin.y - margin},
-                                                {field.getLeftPenaltyLine().end.x + margin, field.getLeftPenaltyLine().end.y + margin},
-                                                {backLineUsXCoordinate, field.getLeftPenaltyLine().end.y + margin},
-                                                {backLineUsXCoordinate, field.getLeftPenaltyLine().begin.y - margin}};
-
-    interface::Input::drawDebugData(defenceAreaUsPoints);
-    Polygon defenceAreaUs(defenceAreaUsPoints);
-
-    std::vector<Vector2> defenceAreaThemPoints = {{field.getRightPenaltyLine().begin.x - margin, field.getRightPenaltyLine().begin.y - margin},
-                                                  {field.getRightPenaltyLine().end.x - margin, field.getRightPenaltyLine().end.y + margin},
-                                                  {backLineThemXCoordinate, field.getRightPenaltyLine().end.y + margin},
-                                                  {backLineThemXCoordinate, field.getRightPenaltyLine().begin.y - margin}};
-
-    Polygon defenceAreaThem(defenceAreaThemPoints);
-    return ourDefenseArea ? defenceAreaUs : defenceAreaThem;
+    double backLineChanges = includeOutSideField ? field.getBoundaryWidth() : margin;
+    Vector2 bottomGoal = ourDefenseArea ? field.getOurBottomGoalSide() + Vector2(-backLineChanges, -margin) : field.getTheirBottomGoalSide() + Vector2(backLineChanges, -margin);
+    Vector2 topGoal = ourDefenseArea ? field.getOurTopGoalSide() + Vector2(-backLineChanges, margin) : field.getTheirTopGoalSide() + Vector2(backLineChanges, margin);
+    Vector2 bottomPenalty = ourDefenseArea ? field.getLeftPenaltyLineBottom() + Vector2(margin, -margin) : field.getRightPenaltyLineBottom() + Vector2(-margin, -margin);
+    Vector2 topPenalty = ourDefenseArea ? field.getLeftPenaltyLineTop() + Vector2(margin, margin) : field.getRightPenaltyLineTop() + Vector2(-margin, margin);
+
+    std::vector<Vector2> defenseArea = {bottomPenalty, topPenalty, topGoal, bottomGoal};
+    interface::Input::drawDebugData(defenseArea);
+    return Polygon(defenseArea);
```

Comment:
```suggestion
    return Polygon{ defenseArea };
```

Reply:
What is the difference? I more like the version where the constructor is called with a vector of positions.